org.artisan_scope.artisan.yml

```diff
@@ -0,0 +1,107 @@
+  # cleanup QtWebEngine dictionaries, which we don't need and use quite some space
+  - find $FLATPAK_DEST/share/locale -name \*.bdic | xargs rm -f
+  - rm -Rf $FLATPAK_DEST/qtwebengine_dictionaries
+  # we need no QtWebEngine developer tools, nor automation driver
+  - rm -f $FLATPAK_DEST/resources/qtwebengine_devtools_resources.pak
+  - rm -Rf $FLATPAK_DEST/libexec/webenginedriver
+  # cleanup base
+  - /app/cleanup-BaseApp.sh
+
+command: artisan
+separate-locales: false
+finish-args:
+  # Qt
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  #- --device=dri # we already have --device=all
+  # USB
+  - --device=all
+  # allow network access for network devices and artisan.plus
+  - --share=network
+  # storing and retrieving artisan.plus credentials
+  - --talk-name=org.freedesktop.secrets
+  # filesystem access needed for autosave
+  - --filesystem=xdg-documents
+  # required for QtWebEngine
+  - --env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess
+
+modules:
+  - ./shared-modules/libusb/libusb.json
+  - ./dep-lib-phidgets.yml
+  - ./dep-lib-qhull.yml
+  - ./dep-python3-wheels.json
+  - ./dep-python3-source.json
+
+  - name: artisan
+    buildsystem: simple
+    build-commands:
+      # install without application bundle with dependencies
+      - sed -i 's/^\s*\(pyinstaller\|rm -rf dist\)/#\1/' src/build-linux.sh
+      - mkdir -p src/dist/artisan && touch src/dist/artisan/__dummy__
+      - cd src && PATH=/var/data/python/bin:$PATH ./build-linux.sh
+      - rm -f $FLATPAK_DEST/__dummy__
+      # copy all gathered files to the destination
+      - mkdir $FLATPAK_DEST/includes
+      - cp -Rp src/dist/* $FLATPAK_DEST/includes
+      # and also the Python files
+      - cp -Rp src/{artisan.py,artisanlib,plus,proto,help,uic,misc} $FLATPAK_DEST
+      - chmod a+x $FLATPAK_DEST/artisan.py
+      # fix translations not being found by default
+      - mv $FLATPAK_DEST/includes/translations/* $FLATPAK_DEST/translations
+      - rmdir $FLATPAK_DEST/includes/translations
+      - sed -i "s/trans_paths\.append('translations')/trans_paths.append('\/app\/translations')/" $FLATPAK_DEST/artisanlib/main.py
+      # then compile all Python files
+      - python3 -m compileall $FLATPAK_DEST/{artisanlib,plus,proto,help,uic,misc}
+      # Flatpak expects Qt plugins in lib/plugins/ instead of plugins/
+      - ln -s ../plugins $FLATPAK_DEST/lib/
+      # desktop integration
+      - ln -s /app/artisan.py /app/bin/artisan
+      - mkdir -p $FLATPAK_DEST/share/{metainfo,applications,icons/hicolor/scalable/apps}
+      - install -Dm644 src/artisan.svg $FLATPAK_DEST/share/icons/hicolor/scalable/apps/$FLATPAK_ID.svg
+      - install -Dm644 src/debian/usr/share/metainfo/$FLATPAK_ID.metainfo.xml $FLATPAK_DEST/share/metainfo/
+      - install -Dm644 src/debian/usr/share/applications/$FLATPAK_ID.desktop $FLATPAK_DEST/share/applications/
+      - sed -i "s/^Icon=.*$/Icon=$FLATPAK_ID/" $FLATPAK_DEST/share/applications/$FLATPAK_ID.desktop
```

Comment:
I'm not that happy with installation procedure. Upstream only makes fully bundled packages, including Qt and friends, which is not so appropriate for Flatpaks. So after some discussion upstream, we decided to have the installation procedure here in the Flatpak, and gradually move (hopefully most of it) to a more integrated installation approach upstream.

Reply:
Use `desktop-file-edit` instead of sed here to change the icon.

Reply:
Thank you, that is much better!